tests/test_page.py

```diff
@@ -10,23 +10,48 @@
 
 
 def test_base_url(base_url, page):
-    assert base_url == page.url
+    assert base_url == page.canonical_url
 
 
-def test_url_keywords(base_url, selenium):
+def test_canonical_url_absolute(base_url, selenium):
+    url_template = 'https://www.test.com/'
+
+    class MyPage(Page):
+        URL_TEMPLATE = url_template
+    page = MyPage(selenium, base_url)
+    assert url_template == page.canonical_url
+
+
+def test_canonical_url_empty(selenium):
+    page = Page(selenium)
+    assert page.canonical_url == ''
+
+
+def test_canonical_url_keywords(base_url, selenium):
```

Comment:
It does seem excessive, especially knowing the implementation, but we shouldn't write tests based on our knowledge of the implementation, so I agree that the extra coverage might be worthwhile in case the implementation changes in the future and a possible bug is introduced. I'll add a test for it.